src/state.js

```diff
@@ -1,12 +1,17 @@
+import document from 'global/document';
 import window from 'global/window';
 
+// Tracks previously-downloaded scripts by URL.
+const scriptCache = new window.Map();
```

Comment:
actually, I think this can probably just be a `Set`, right? We don't actually have values other than keys. Has same support as Map.